### PR TITLE
[Merged by Bors] - feat(Data/Nat/Basic): self_add_sub_one and friends

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Basic.lean [hunk 1]
```diff
@@ -327,6 +327,21 @@ theorem le_of_pred_lt {m n : ℕ} : pred m < n → m ≤ n :=
...
+    convert (add_succ_sub_one (Nat.succ _) _).symm
+
+lemma sub_one_add_self (n : ℕ) : (n - 1) + n = 2 * n - 1 :=
+  add_comm _ n ▸ self_add_sub_one n
+
+lemma self_add_pred (n : ℕ) : n + pred n = 2 * n - 1 :=
+  self_add_sub_one n
+
+lemma pred_add_self (n : ℕ) : pred n + n = 2 * n - 1 :=
+  sub_one_add_self n
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Basic.lean:343]:
```suggestion

lemma self_add_pred (n : ℕ) : n + pred n = (2 * n).pred :=
  self_add_sub_one n

lemma pred_add_self (n : ℕ) : pred n + n = (2 * n).pred :=
  sub_one_add_self n
```
feels more consistent here.

Comment (mo271) [reply Mathlib/Data/Nat/Basic.lean]: of course...

File: Mathlib/Data/Nat/Basic.lean [hunk 2]
```diff
@@ -327,6 +327,21 @@ theorem le_of_pred_lt {m n : ℕ} : pred m < n → m ≤ n :=
...
+  · rw [two_mul]
+    convert (add_succ_sub_one (Nat.succ _) _).symm
+
+lemma sub_one_add_self (n : ℕ) : (n - 1) + n = 2 * n - 1 :=
+  add_comm _ n ▸ self_add_sub_one n
+
+lemma self_add_pred (n : ℕ) : n + pred n = (2 * n).pred :=
+  self_add_sub_one n
+
+lemma pred_add_self (n : ℕ) : pred n + n = (2 * n).pred :=
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Basic.lean:342]:
```suggestion
theorem pred_add_self (n : ℕ) : pred n + n = (2 * n).pred :=
```
throughout, to match the rest of the file

bors d+

File: Mathlib/Data/Nat/Basic.lean [hunk 3]
```diff
@@ -327,6 +327,21 @@ theorem le_of_pred_lt {m n : ℕ} : pred m < n → m ≤ n :=
...
+  cases n
+  · rfl
+  · rw [two_mul]
+    convert (add_succ_sub_one (Nat.succ _) _).symm
+
+lemma sub_one_add_self (n : ℕ) : (n - 1) + n = 2 * n - 1 :=
+  add_comm _ n ▸ self_add_sub_one n
+
+lemma self_add_pred (n : ℕ) : n + pred n = (2 * n).pred :=
+  self_add_sub_one n
```
Comment (mo271) [inline Mathlib/Data/Nat/Basic.lean:340]:
```suggestion
theorem self_add_sub_one (n : ℕ) : n + (n - 1) = 2 * n - 1 := by
  cases n
  · rfl
  · rw [two_mul]
    convert (add_succ_sub_one (Nat.succ _) _).symm

theorem sub_one_add_self (n : ℕ) : (n - 1) + n = 2 * n - 1 :=
  add_comm _ n ▸ self_add_sub_one n

theorem self_add_pred (n : ℕ) : n + pred n = (2 * n).pred :=
  self_add_sub_one n
```

**Pull request closed without merging**